### PR TITLE
fix(cli): stop workspace create from orphaning its key, self-heal workspace active

### LINKED ISSUE #1260: workspace resolve 404: `workspace create` bypasses Cloud, orphaning the key it stores
## Summary

`agent-relay workspace create` / `agentworkforce login --workspace <key>` mints a workspace directly against Relaycast and never creates the corresponding Cloud Postgres row, so `agent-relay workspace active` / `agentworkforce deploy` immediately 404s ("Workspace not found") on that same key. This is the recurring "dev vs prod workspace" resolve failure seen across `cloud`, `agents`, `workforce`, and `internal-agents` sessions since at least 2026-07-04.

Reproduced today in `internal-agents`:
```
agentworkforce deploy market-intel/reddit-monitor/persona.json --mode cloud --on-exists update --input SLACK_CHANNEL=C0ADJH54N3T
agentworkforce deploy failed: Workspace resolve failed at /api/v1/workspaces/active?key=rk_live_c590d2bc...: 404 Workspace not found
```
The failing key is the locally cached `default` workspace key in `~/.agentworkforce/relay/workspaces.json` — it was never created through Cloud's unified endpoint, so it has no `relay_workspaces` row to resolve.

## Root cause (confirmed via live reproduction 2026-07-13, full report below)

`agent-relay workspace create` creates a messaging-only workspace directly in Relaycast (`RelayCast.createWorkspace`, `relay/packages/sdk/src/agent-relay.ts:127-150`) and stores only the returned `rk_live_` key locally. `agent-relay workspace active` (used by `agentworkforce login`/`deploy`) resolves that key through Cloud's `resolveActiveWorkspace` (`relay/packages/cloud/src/workspaces.ts:342-379`), whose primary lookup is `getRelayWorkspaceByRelaycastApiKey` against Cloud Postgres. The direct-Relaycast creation path never inserts that row, so Cloud correctly 404s even though the key is valid in Relaycast (`GET https://gateway.relaycast.dev/v1/workspace` returns 200 for the same key).

This mismatch was introduced by commit `0f87d23be`, which added the Cloud-backed `workspace active` command alongside the pre-existing direct-Relaycast `workspace create` command without updating `create` or adding a create→active integration test.

Confirmed **not** the same bug as #2561/#2565: those fix row drift/partial-failure *inside* Cloud's unified create path. A key created via `agent-relay workspace create` never enters that path at all, so there's no row for #2565's reconciliation to act on. Live positive control: Cloud's own unified create→resolve returns 201→200 (proving #2565 is deployed and working), while a fresh `agent-relay workspace create` key remains Relaycast-valid and Cloud-404 afterward.

## Fix recommendation

Primary fix belongs in `AgentWorkforce/relay`:

1. Change `relay/packages/cli/src/cli/commands/workspace.ts` so `workspace create` calls `@agent-relay/cloud.createWorkspace` (the unified Cloud endpoint), not `AgentRelay.createWorkspace`/`RelayCast.createWorkspace` (direct Relaycast-only).
2. Extend `WorkspaceCreateResponse` / `normalizeWorkspaceCreateResponse` (`relay/packages/cloud/src/types.ts`, `workspaces.ts`) to retain the Cloud response's `relaycastApiKey`, and store that via `setWorkspaceKey(name, key)`.
3. Align the CLI option name with Cloud (`--api-url`) rather than the current Relaycast-only `--base-url`; fail clearly when no Cloud session exists.
4. Add a regression test at the CLI boundary: authenticated `workspace create` must call Cloud `/api/v1/workspaces`, save its returned Relaycast key, and an immediate `workspace active --json` must resolve it successfully. Assert it does NOT invoke direct `RelayCast.createWorkspace`.
5. Keep `AgentRelay.createWorkspace` available as an explicit messaging-only SDK primitive if that's still wanted publicly — the bug is using it for a CLI command whose stored workspace is later advertised as canonical Cloud/Relayfile state.

### Existing orphaned keys

Fixing the CLI prevents new orphans but doesn't repair keys already stored locally (e.g. this machine's `default`, `mini-sandbox-team`, and others in `~/.agentworkforce/relay/workspaces.json`). Needs a separate, explicitly-authenticated migration/adoption command that validates the presented key via Relaycast, provisions a canonical Cloud/Relayfile workspace, and persists both the plaintext key and the Relaycast provider workspace ID under the authenticated Cloud user. Should not silently auto-create rows from an unauthenticated resolver request.

## Repository ownership

| Repository | Role | Verdict |
|---|---|---|
| `relay` | Direct create + local key store + Cloud-backed active resolver | **Primary fix owner** |
| `cloud` | Canonical unified create/resolve; optional orphan-adoption endpoint | Working correctly for its own create path |
| `relaycast-cloud` | Messaging workspace/key creation and auth | Healthy; not the missing-row owner |
| `relayfile` | Calls `workspace active`, then requests delegated credentials | Downstream victim; its own `setup` path already uses the correct Cloud create |
| `relayfile-cloud` | Relayfile data plane, post-resolution | Not involved |

## Full investigation report

A full reproduction (curl traces against all 3 resolver candidates, live positive/negative controls, exact file:line references) is available — happy to attach/paste on request. Investigated 2026-07-13 by an autonomous relay investigation session (`019f5b95-8749-7da0-a54b-8c5f4678d8fe`); reproduced again independently today (2026-07-14) hitting the exact same failure from `internal-agents`.

### PR DESCRIPTION
**Pull request by khaliqgant**
## Summary

Fixes #1260 — `agent-relay workspace create` (used by `agentworkforce login --workspace <key>`) minted a workspace directly against Relaycast and never created the corresponding Cloud Postgres row, so `workspace active` / `agentworkforce login` / `deploy` immediately 404 with "Workspace not found" on that same key, forever. Root-caused with a live reproduction in #1260.

## What changed

- **`workspace create` now goes through Cloud's unified `/api/v1/workspaces` endpoint** (`@agent-relay/cloud.createWorkspace`, the same path `relayfile setup` already uses) instead of the direct-Relaycast `AgentRelay.createWorkspace`. That endpoint persists the missing Postgres row, so the returned key actually resolves. Renamed the command's `--base-url` flag to `--api-url` to match `workspace active`'s existing flag (it now points at Cloud, not the Relaycast gateway). Fails loudly instead of silently storing an unusable key if the response is missing `relaycastApiKey`.
- **`resolveActiveWorkspace` self-heals.** It opportunistically caches the resolved `cloudWorkspaceId` alongside each local workspace alias. If a key later stops resolving (rotated server-side, or a pre-fix orphan created before this PR), it re-mints a working key for that *same* workspace via the session-authenticated `POST /api/v1/workspaces/{id}/join` endpoint and retries once — no new workspace is ever created. Best-effort: any failure (no session, no longer a member, no cached id to heal from) falls through to the original resolve error, so the failure mode is unchanged for cases self-heal can't help with.

## Why this shape

- The `/join` endpoint already exists and is exactly the right primitive — session-authenticated, returns the canonical `relaycastApiKey` for a workspace the caller already belongs to. No new server-side (cloud repo) code was needed.
- Self-heal only fires when there's a cached `cloudWorkspaceId` from a prior successful resolve, so it can't paper over a workspace the CLI has never proven the caller has access to.

## Test plan

- [x] `packages/cloud/src/workspace-store.test.ts` — caching/preserving `cloudWorkspaceId` across key updates
- [x] `packages/cloud/src/workspaces.test.ts` — caches `cloudWorkspaceId` on success; self-heals via `joinWorkspace` on a stale key; falls through to the original error with no cached id; falls through when self-heal itself fails; `joinWorkspace` unit tests
- [x] `packages/cli/src/cli/commands/workspace.test.ts` — `create` stores the Cloud-issued key and prints it; fails loudly when the key is missing
- [x] `packages/cli/src/cli/lib/sdk-command.test.ts` (new) — asserts `withSdkDefaults().createWorkspace` calls `@agent-relay/cloud`, never the direct-Relaycast SDK
- [x] `npx vitest run packages/cloud packages/cli` — 776 passed, 9 pre-existing unrelated failures (confirmed present on `main` before this branch too), 0 regressions
- [x] `npx tsc --noEmit` clean on both `packages/cloud` and `packages/cli` for every file this PR touches

🤖 Generated with [Claude Code](https://claude.com/claude-code)

<a href="https://cubic.dev/pr/AgentWorkforce/relay/pull/1261?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

**Comment by khaliqgant** [comment]
Addressed the review feedback in the latest commit (26be7b33d):

**cubic-dev-ai's 4 issues** — all fixed:
1. `workspace-store.ts:79` — `setWorkspaceKey` no longer carries a stale `cloudWorkspaceId` forward on a plain key update; only an explicitly-supplied id is cached now. A user repointing an alias at a different workspace via `set_key`/`join` can no longer have self-heal silently rejoin the old workspace.
2. `commands/workspace.ts:69` — `workspace create` now passes `workspace.workspaceId` to `setWorkspaceKey` immediately from the create response, instead of waiting for the first successful resolve.
3. `workspaces.ts` (`tryPostJson`) — now forwards `interactive`/`refreshTimeoutMs` into `authorizedApiFetch` as its 4th arg, matching `tryGetJson`'s existing pattern, so a 401 mid-request during self-heal's `/join` call can't trigger an unwanted interactive browser re-auth.
4. `workspaces.ts:443` — `resolveActiveWorkspace` now wraps the primary `resolveWorkspaceDescriptor` call in a catch, folding a thrown non-404/405 error (e.g. 401/403 from a rotated/invalid key) into the same `{ lastUnsupported }` shape, so self-heal still gets a chance instead of the error bypassing it entirely.

**gemini-code-assist** flagged the same #4 independently — same fix covers it.

**coderabbitai nitpicks** — both addressed (same underlying fix as #1/#2 above); updated the corresponding test assertion.

New regression test added for the 401-bypasses-self-heal case, plus the `setWorkspaceKey` clearing behavior. Full diff: `npx vitest run packages/cloud packages/cli` — 777 passed, only 2 pre-existing unrelated failures (confirmed present on `main` before this branch too). `tsc --noEmit` clean on both packages.

File: packages/cloud/src/workspaces.ts [hunk 1]
```diff
@@ -373,8 +395,107 @@ export async function resolveActiveWorkspace(
...
+    );
+  }
+  const { name, entry } = active;
+
+  // A non-404/405 failure (e.g. 401/403 from a rotated/invalid key) throws
+  // out of resolveWorkspaceDescriptor instead of returning `{ lastUnsupported }`
+  // — catch it here and fold it into the same shape so self-heal below still
+  // gets a chance whenever a cached cloudWorkspaceId is available, instead of
+  // this propagating straight past the self-heal block.
+  const primary = await resolveWorkspaceDescriptor(entry.key, options).catch(
```
[the remark this answers is not in the document]
Comment (khaliqgant) [reply packages/cloud/src/workspaces.ts:478]: Fixed in 5731c2a06. Added a `WorkspaceResolveHttpError` thrown only for a genuine non-2xx HTTP response from the resolve endpoint itself, and narrowed `resolveActiveWorkspace`'s catch to only that type via `instanceof` — a network/transport failure, an `ensureAuthenticated` session error, or a malformed-response schema error now rethrows unchanged instead of being folded into `{ lastUnsupported }` and triggering self-heal. Added a regression test proving a transport-level fetch failure rejects with the original error, never calls `/join`, and never touches the stored key. (Also had to switch the existing 401 self-heal test to 403 — turns out a 401 makes `authorizedApiFetch` attempt a token refresh + retry first, which the test wasn't mocking; that's an orthogonal concern from what this fix addresses.)
